loopy/__init__.py

```diff
@@ -487,6 +487,48 @@ def make_copy_kernel(new_dim_tags, old_dim_tags=None):
 # }}}
 
 
+# {{{ einsum
+
+def make_einsum(spec, arg_names, knl_creation_kwargs=None):
+    arg_spec, out_spec = spec.split("->")
+    arg_specs = arg_spec.split(",")
+
+    out_indices = set(out_spec)
```

Comment:
Hey look.. that's an IndexLambda. :)

Reply:
Yeah. I fear pytato is going to grow its own reimplementation of this. (Unless we make the core of it reusable somehow? The different sum node types would be one thing we'd need to work around.)